src/solvers/dgsem_p4est/dg_2d.jl

```diff
@@ -20,10 +20,24 @@ function create_cache(mesh::P4estMesh{2}, equations, mortar_l2::LobattoLegendreM
 end
 
 
-# TODO: p4est interface performance, move and generalize this function for 3D
-@inline function index_to_start_step(index::Symbol, index_range)
+#     index_to_start_step_2d(index::Symbol, index_range)
+#
+# Given a symbolic `index` and an `indexrange` (usually `eachnode(dg)`),
+# return an index value to begin a loop and an index step to update during a
+# loop. The resulting indices translate surface indices to volume indices.
+#
+# !!! warning
+#     This assumes that loops using the return values are written as
+#
+#     my_i, my_j = # begin values
+#     for i in ...
+#       # do stuff with i and (my_i, my_j)
+#       my_[ij] += my_[ij]_step
+#     end
```

Comment:
Thanks for including an example. Unfortunately, I do not understand it :sob: 
Which variables are the return values from `index_to_start_step`? What is `my_[ij]`? Is this `ij` and index to the array `my_`? What is in the array `my_`? :confused:

Reply:
I tried to improve the documentation of these new auxiliary functions.

Reply:
Thanks, much much clearer now imho!